File: 0881-boats-to-save-people/0881-boats-to-save-people.py
class Solution:
    def numRescueBoats(self, people: List[int], limit: int) -> int:
        n=len(people)
        people.sort()
        i=0
        j=n-1
        boats=0
        while i<=j:
            weight = people[i]+people[j]
            if weight > limit:
                boats+=1
                j-=1
            else: 
                i+=1 
                j-=1
                boats+=1
        return boats

File: 0881-boats-to-save-people/0881-boats-to-save-people.cpp
class Solution {
public:
    int numRescueBoats(vector<int>& people, int limit) {
        int n = people.size();
        sort(people.begin(), people.end());
        int i=0;
        int j=n-1;
        int boats=0;
        while (i<=j){
            int weight = people[i] + people[j];
            if (weight > limit){
                boats++;
                j--;
            }
            else{
                i++;
                j--;
                boats++;
            }
        }
        return boats;
    }
};